query/src/main/java/org/infinispan/query/impl/massindex/DistributedExecutorMassIndexer.java

```diff
@@ -57,6 +59,68 @@ public void start() {
       return executeInternal(true);
    }
 
+   private void addFutureListToFutures(List<CompletableFuture<Void>> futures, List<CompletableFuture<Void>> futureList) {
+      futureList.forEach(f -> futures.add(f.exceptionally(t -> {
+         if (t instanceof InterruptedException) {
+            Thread.currentThread().interrupt();
+            return null;
+         } else if (t instanceof CompletionException) {
+            Throwable cause = t.getCause();
+            throw LOG.errorExecutingMassIndexer(cause);
+         } else {
+            throw LOG.errorExecutingMassIndexer(t);
+         }
+      })));
+   }
+
+   @Override
+   public CompletableFuture<Void> reindex(Object... keys) {
+      List<CompletableFuture<Void>> futures = new ArrayList<>();
+      Set<Object> everywhereSet = new HashSet<>();
+      Set<Object> primeownerSet = new HashSet<>();
+      for (Object key : keys) {
+         Class<?> indexedType = cache.get(key).getClass();
```

Comment:
I just want the developer aware of the key not exist by the NPE.
Anyway, I just modify it as your wish.